### PR TITLE
Adds support for mapping location from fedora.

### PR DESCRIPTION
**Pull request by justinlittman**
refs #1412

## Why was this change made?
Add support for mapping location from fedora.

## How was this change tested?
Unit, sdr-deploy

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Comment by justinlittman** [comment]
Before:
```
Status (n=100000):
  Success:   36579 (36.579%)
  Different: 62769 (62.769%)
  To Cocina error:     41 (0.041%)
  To Fedora error:     1 (0.001%)
  Missing:     610 (0.61%)
```

After:
```
Status (n=100000):
  Success:   36579 (36.579%)
  Different: 62745 (62.745%)
  To Cocina error:     41 (0.041%)
  To Fedora error:     25 (0.025%)
  Missing:     610 (0.61%)
```

The additional Fedora errors are fixed by https://github.com/sul-dlss/dor-services-app/pull/1443 so OK to ignore.

File: app/services/cocina/from_fedora/descriptive/location.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  module FromFedora
+    class Descriptive
+      # Maps locations
+      class Location
+        PURL_REGEX = %r{^https?://purl.stanford.edu/}.freeze
+
+        # @param [Nokogiri::XML::Element] resource_element mods or relatedItem element
+        # @param [Cocina::FromFedora::Descriptive::DescriptiveBuilder] descriptive_builder
+        # @return [Hash] a hash that can be mapped to a cocina model
+        def self.build(resource_element:, descriptive_builder: nil)
```
Comment (mjgiarlo) [inline app/services/cocina/from_fedora/descriptive/location.rb:13]: The `descriptive_builder` param doesn't seem to be used. Cruft? Or placeholder for later?

**Review by mjgiarlo — APPROVED** [review]
Non-blocking question. Merge at will.

File: app/services/cocina/from_fedora/descriptive/location.rb (hunk 1 above)
Comment (justinlittman) [reply app/services/cocina/from_fedora/descriptive/location.rb:13]: It's part of the interface that all builders must implement.

Comment (mjgiarlo) [reply app/services/cocina/from_fedora/descriptive/location.rb:13]: :beers:

**Pull request merged**